src/schema/game_state.py

```diff
@@ -113,6 +117,7 @@ def is_onboarding_complete(self) -> bool:
             self.player is not None
             and self.player.is_onboarding_complete()
             and self.chat_history_for_onboarding_complete
+            and self.onboarding_agent_has_completed
```

Comment:
suggestion (non-blocking): If we have the explicit flag, do we want to just check that exclusively?  ANDing the others seems to only leave the door open for new strange cases.

Reply:
Yeah I think that feels right - i'll make a note to add a cleanup pr for it